src/urllib3/contrib/emscripten/connection.py

```diff
@@ -0,0 +1,216 @@
+from __future__ import annotations
+
+import os
+import typing
+from http.client import ResponseNotReady
+
+from ..._base_connection import _TYPE_BODY
+from ...connection import HTTPConnection, ProxyConfig, port_by_scheme
+from ...exceptions import ResponseError, TimeoutError
+from ...response import BaseHTTPResponse
+from ...util.connection import _TYPE_SOCKET_OPTIONS
+from ...util.timeout import _DEFAULT_TIMEOUT, _TYPE_TIMEOUT
+from ...util.url import Url
+from .fetch import _RequestError, _TimeoutError, send_request, send_streaming_request
+from .request import EmscriptenRequest
+from .response import EmscriptenHttpResponseWrapper
+
+
+class EmscriptenHTTPConnection(HTTPConnection):
```

Comment:
Would it be possible to inherit from `BaseHTTPConnection` instead?

Reply:
Can do. I only did it this way because it reduced the amount of messing with the existing codebase. No problems switching it.

Reply:
Can you clarify what amount of messing with the code base we're talking about?

Reply:
I did it, and actually it seems I only had to change one bit in the end - in BaseHttpResponse, where something expects an HTTPConnection, not a BaseHTTPConnection